.github/workflows/generate-chain-specs.yaml

```diff
@@ -0,0 +1,82 @@
+name: Generate Chain Specs
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - 'chain-spec-generator/**'
+      - 'relay/**'
+      - 'system-parachains/**'
+      - 'Cargo.toml'
+      - 'Cargo.lock'
+
+permissions:
+  contents: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  generate-chain-specs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y git protobuf-compiler
+
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: 1.79.0
+          target: wasm32-unknown-unknown
+          components: rust-src
+          override: true
+
+      - name: Add rust-src
+        run: rustup component add rust-src --toolchain 1.79.0-x86_64-unknown-linux-gnu
```

Comment:
Same than above. Let's use 1.77 as the [fellowship does](https://github.com/polkadot-fellows/runtimes/releases/tag/v1.3.2).